de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/editor/LocalEditorHandler.java

```diff
@@ -82,9 +79,9 @@ public Editor openEditor(
             return null;
         }
 
-        SPath path = virtualFileConverter.convertToPath(virtualFile);
+        SPath path = VirtualFileConverter.convertToSPath(virtualFile);
 
-        if (path == null) {
+        if (path == null || !SessionUtils.isShared(path)) {
             LOG.debug("Ignored open editor request for file " + virtualFile +
```

Comment:
I guess you can now distinguish between a shared resource and a resource that does not exist ?

Reply:
Yes, since I am no longer reliant on the session (or rather the IProjects held by the session). Beforehand, I just used the project objects held by the session and iterated over them to determine whether the given resource belonged to one of them. If it did, I used the IProject object to convert the resource. This approach stems from the old IProject implementation (before the V2 rework) that held an internal state, meaning just creating a new IProject object on the fly was not possible. A result of this implementation was, that it was not possible to determine whether a resource did not exist or was simply not part of one of the shared modules.

Since this is no longer the case (IProject objects are now wrappers and don't hold any real state), the new implementation now uses the IntelliJ API to determine the module the resource belongs to and creates a matching IProject object on the go. This allows the us to create IResources for every resource belonging to a shareable module (since we still need an IProject object and the check whether a module meets our restrictions is included in the IntellijProjectImplV2 CTOR).